js/src/common/utils/extractText.ts

```diff
@@ -1,15 +1,14 @@
+import Mithril from 'mithril';
+
 /**
  * Extract the text nodes from a virtual element.
- *
- * @param {VirtualElement} vdom
- * @return {String}
  */
-export default function extractText(vdom) {
+export default function extractText(vdom: Mithril.Children): string {
   if (vdom instanceof Array) {
     return vdom.map((element) => extractText(element)).join('');
   } else if (typeof vdom === 'object' && vdom !== null) {
-    return vdom.children ? extractText(vdom.children) : vdom.text;
+    return vdom.children ? extractText(vdom.children) : (vdom.text as string);
   } else {
-    return vdom;
+    return vdom as string;
```

Comment:
Same here. At this point, `vdom` is `string | number | boolean | null | undefined`. An explicit match would be safer, otherwise there's not much benefit from this typing.

Reply:
Do we want to cast it or have `extractText` return `string | number | boolean | null | undefined`, then?

Casting would technically be change in behavior, but I don't know what the use case of that behavior would be.

Reply:
`extractText` implies that it returns text. I think casting it would be best.